### ISSUE TITLE
Get on latest/characters/[character ID]/contacts/ results in 500 error

### ISSUE THREAD

**Issue by janhkrueger** [ccpgames/esi-issues#319]
Reading contact list with NPCs/NPC corps as contacts result in a 500 error.

Command used:

`curl -i --compressed -X GET --header 'Accept: application/json' --header 'Authorization: Bearer [access token]' 'https://esi.tech.ccp.is/latest/characters/[character ID]/contacts/?datasource=tranquility&page=1'`

Gives an 

```
HTTP/1.1 500 Internal Server Error 
{"error": "Action failed"}
```

Doings first described here:
https://gist.github.com/janhkrueger/a4989e9d7764bba470f73aa32f62e23f

Related topic in the EVE forums:
https://forums.eveonline.com/default.aspx?g=posts&t=514449&find=unread

**Comment by Telac-DK** [comment]
Same issues here. Some contact calls ended in error. Today all contacts write fails. Not just you :-)

**Comment by Telac-DK** [comment]
I opened Eve, and checked contact list. I had a strange looking contact with white circle as standing and no name. After opening it in window, its name was #System, so I looked that up in Eve db export ItemsType table, and found out this was item id 0.

Since I couldn't delete this contact from within Eve, I tried the swagger website, and called a contacts delete for id 0, and voila, it deleted it.

Since then, all my API calls to contacts no longer returns error 500 Action failed.

It should not be possible to add contact id's that are not corp/alliance/persons, but it seems like its possible. Try check your contacts for NPC corps, funny items etc within Eve, that could be the culprit

**Comment by janhkrueger** [comment]
Interesting.
Will check it when beeing back at home.

**Comment by Telac-DK** [comment]
I had it multiple times now (when accidently adding wrong ids in standings api), and most often EVE can't even open the standings list anymore. Since you no longer can read out the id's via API, I have started to save whatever id is sent to EVE, and slowly removing the contacts one by one untill I found the culprit.

Maybe you should try removing all your contacts and try again? (If you can delete them in Eve that is)

CCP accepts blindly anything as a contact, so I have started checking any contact I intend to add, via the API for getting a corp name, to ensure its a valid target before screwing up my list.

**Comment by janhkrueger** [comment]
I only have contacts added via the game client. No other tool or API involved. Not even EVE Gate.
Indeed I have checked every of my contacts in game yesterday, this is why I removed the NPC corp "Arkombine".

Also: I will not say to every of my intended users that they have to check their contacts. The endpoint has to work in a reliable way. Removing and adding all contacts again is something a single user can do. Something I can do to find the cause in this case.
Nothing I can say to every possible user.

When there're contacts who cannot returned with this endpoint for whatever reason, I can live when the endpoint give the info with a special error code so a player can take a deeper look into his/her characters contacts.
This issue has to solved in the endpoint. Not on the user side.
